.github/workflows/build-and-cache.yml

```diff
@@ -0,0 +1,24 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  check:
+    runs-on: ubuntu-22.04
+    permissions:
+      id-token: "write"
+      contents: "read"
+    steps:
+      - uses: actions/checkout@v4
+      - uses: DeterminateSystems/nix-installer-action@main
+      - uses: DeterminateSystems/magic-nix-cache-action@main
+      - uses: DeterminateSystems/flake-checker-action@main
+      - uses: cachix/cachix-action@v14
+        with:
+          # https://nix.dev/guides/recipes/continuous-integration-github-actions
+          name: superkenvery
+          authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
```

Comment:
I suppose i'd need to register on cachix?
probably also adjust based on that info. (e.g. add cachix token to the list of variables i guess)

Reply:
Yes you would need to register on cachix, if you want to use cachix as a binary cache.

There are also other options, including some self-host ones, if you don't like cachix. But AFAIK cachix is the most convenient and popular one.

Reply:
For information on how to setup the github action and get it running, [here](https://nix.dev/guides/recipes/continuous-integration-github-actions) is a good guide.